src/v/wasm/allocator.cc

```diff
@@ -48,7 +50,8 @@ std::optional<heap_memory> heap_allocator::allocate(request req) {
     return front;
 }
 
-void heap_allocator::deallocate(heap_memory m) {
+void heap_allocator::deallocate(heap_memory m, size_t used_amount) {
+    std::memset(m.data.get(), 0, std::min(used_amount, m.size));
```

Comment:
defensive programming and all, yay. but it also reads as if more could have been used than the buffer size. is that a thing? maybe used_amount corresponds to an amount that perhaps spans instances of heap_memory? that would explain it i think

Reply:
You're right, I should just remove the `std::min` it's overly defensive